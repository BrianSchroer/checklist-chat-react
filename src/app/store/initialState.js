import * as modalDialogType from '../modalDialogType';

export default {
    rooms: [],
    currentRoomId: null,
    ajaxCallsInProgressCount: 0,
    modalDialog: modalDialogType.NONE
};
