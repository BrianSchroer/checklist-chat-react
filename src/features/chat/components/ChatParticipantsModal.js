import React, {PropTypes} from 'react';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';
import * as modalDialogType from '../../../app/modalDialogType';
import {requestChatParticipantsModalDialog, hideModalDialog} from '../../../app/modalDialogDucks';

class ChatParticipantsModal extends React.Component {
    constructor(props, context) {
        super(props, context);
        this.state = {shouldDisplayModal: false};

        this.resetState = this.resetState.bind(this);
        this.closeModal = this.closeModal.bind(this);
    }

    componentWillReceiveProps(nextProps) {
        this.resetState(nextProps);
    }

    resetState(props) {
        this.setState({
            shouldDisplayModal: props.shouldDisplayModal,
            chatParticipants: props.chatParticipants
        });
    }

    closeModal(event) {
        event.preventDefault();
        this.props.actions.hideModalDialog();
    }

    participantRow(participant) {
        return(
            <tr>
                <td>{participant.name}</td>
                <td>{participant.department}</td>
                <td>{participant.title}</td>
                <td>{participant.connection}</td>
            </tr>
        );
    }

    render() {
        const state = this.state;

        if (!state.shouldDisplayModal) {
            return null;
        }

        return (
            <div className="checklist-chat-modal-backdrop" tabIndex="-1" role="dialog">
                <div className="modal-dialog checklist-chat-modal-dialog" role="document">
                    <div className="modal-content">
                        <div className="modal-header bg-primary text-white">
                            <button type="button" className="close" aria-label="Close" onClick={this.closeModal}>
                                <span aria-hidden="true">&times;</span>
                            </button>
                            <h4 className="modal-title">Who's here?</h4>
                        </div>

                        <div className="modal-body">
                            <table className="table">
                                <thead>
                                    <tr>
                                        <th>Name</th>
                                        <th>Department</th>
                                        <th>Title</th>
                                        <th>Connection</th>
                                    </tr>
                                </thead>
                                <tbody>
                                {this.props.chatParticipants.map((participant, index) =>
                                    (<tr key={index}>
                                        <td>{participant.name}</td>
                                        <td>{participant.department}</td>
                                        <td>{participant.title}</td>
                                        <td>{participant.connection}</td>
                                    </tr>)
                                )}
                                </tbody>
                            </table>
                        </div>

                        <div className="modal-footer">
                            <input type="button" value="Close" className="btn btn-primary" onClick={this.closeModal}/>
                        </div>
                    </div>
                </div>
            </div>
        );
    }
}

ChatParticipantsModal.propTypes = {
    shouldDisplayModal: PropTypes.bool.isRequired,
    chatParticipants: PropTypes.arrayOf(PropTypes.object).isRequired,
    actions: PropTypes.object.isRequired
};

function mapStateToProps(state) {
    const shouldDisplayModal =
        (state.modalDialogRequest && state.modalDialogRequest.type === modalDialogType.CHAT_PARTICIPANTS);

    //const roomId = state.roomId;

    const chatParticipants = [
        {
            name: 'Brian Schroer',
            department: 'Web Development Team 2C',
            title: 'Senior Web Developer',
            connection: 'connection description'
        }
    ];

    return {shouldDisplayModal, chatParticipants};
}

const mapDispatchToProps = (dispatch) => ({
    actions: bindActionCreators(
        {requestChatParticipantsModalDialog, hideModalDialog},
        dispatch)
});

export default connect(mapStateToProps, mapDispatchToProps)(ChatParticipantsModal);
