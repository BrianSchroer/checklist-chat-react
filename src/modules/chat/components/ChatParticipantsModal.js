import React, {PropTypes} from 'react';
import {connect} from 'react-redux';
import Modal from '../../../components/Modal';

export class ChatParticipantsModal extends React.Component {
    constructor(props, context) {
        super(props, context);
    }

    render() {
        const props = this.props;

        const buttons = (
            <input type="button" value="Close" className="btn btn-primary"
                onClick={props.onCloseRequest}/>
        );

        return (
            <Modal title="Who's Here?" onCloseRequest={props.onCloseRequest} buttons={buttons}>
                <div className="chat-participants">
                    <table className="table">
                        <thead>
                            <tr>
                                <th>Name</th>
                                <th>Department</th>
                                <th>Title</th>
                                <th>Connection</th>
                            </tr>
                        </thead>
                        <tbody>
                        {this.props.chatParticipants.map((participant, index) =>
                            (<tr key={index}>
                                <td>{participant.name}</td>
                                <td>{participant.department}</td>
                                <td>{participant.title}</td>
                                <td>{participant.connection}</td>
                            </tr>)
                        )}
                        </tbody>
                    </table>
                </div>
            </Modal>
        );
    }
}

ChatParticipantsModal.propTypes = {
    chatParticipants: PropTypes.arrayOf(PropTypes.object).isRequired,
    onCloseRequest: PropTypes.func.isRequired
};

function mapStateToProps(state, ownProps) {
    let chatParticipants = state.chatParticipants;
    const onCloseRequest = ownProps.onCloseRequest;

    return {chatParticipants, onCloseRequest};
}

export default connect(mapStateToProps)(ChatParticipantsModal);
