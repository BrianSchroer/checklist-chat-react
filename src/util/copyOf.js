export default original => Object.assign({}, original);
