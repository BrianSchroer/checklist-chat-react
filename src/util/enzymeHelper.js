const enzymeHelper = class {

    static nestedFind(wrapper, selector) {
        const selectors = selector.split('>').map(node => node.trim());
        let parent = wrapper;
        let found = [];

        for (let i = 0; i < selectors.length; i++) {
            found = parent.find(selectors[i]);
            if (!found.length) {
                break;
            }
        }

        return found;
    }

    static findSingle(wrapper, selector) {
        const findArray = enzymeHelper.nestedFind(wrapper, selector);

        if (findArray.length != 1) {
            throw(`expected 1 find for selector "${selector}", but found ${findArray.length}.`);
        }

        return findArray.first();
    }
};

export default enzymeHelper;
