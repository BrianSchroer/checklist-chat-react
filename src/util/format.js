import moment from 'moment';

export default class {
    static time(input = '') {

        const isoDateFormatRegex = /\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}([+-]\d{2}:\d{2})?/g;
        let formatted = input;

        if (input && input.match(isoDateFormatRegex)) {
            formatted = this.dateTimeString(input, 'h:mm a');
        }

        return formatted;
    }

    static dateTimeString(input, formatString) {
        let formatted = input;

        if (input) {
            const testMoment = moment(input);

            if (testMoment.isValid()) {
                const testFormat = testMoment.format(formatString);
                if (testFormat != 'Invalid date') {
                    formatted = testFormat;
                }
            }
        }

        return formatted;
    }
}
