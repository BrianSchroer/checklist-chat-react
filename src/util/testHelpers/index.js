import {shallow, mount} from 'enzyme';
import enzymeHelper from './enzymeHelper';

export {
    shallow as shallow,
    mount as mount,
    enzymeHelper as enzymeHelper
};
