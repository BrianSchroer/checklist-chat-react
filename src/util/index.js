import format from './format';
import uiHelper from './uiHelper';
import validationHelper from './validationHelper';

export {
    format as format,
    uiHelper as uiHelper,
    validationHelper as validationHelper
};
