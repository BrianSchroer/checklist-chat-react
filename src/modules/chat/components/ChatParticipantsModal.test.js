import React from 'react';
import initialState from '../../../app/store/initialState';
import {snapshotHelper} from '../../../util/testHelpers';
import {ChatParticipantsModal, mapStateToProps} from './ChatParticipantsModal';

const defaultProps = {
    chatParticipants: [1, 2, 3, 4, 5].map(i => ({
            name:  `name ${i}`,
            department: `department ${i}`,
            title: `title ${i}`,
            connection: `connection ${i}`
        })),
    onCloseRequest: () => {},
    actions: {}
};

const defaultOwnProps = {
    onCloseRequest: () => {}
};

function overrideProps(propOverrides) {
    return Object.assign({}, defaultProps, propOverrides);
}

function assertSnapshotMatch(propOverrides = {}) {
    snapshotHelper.assertMatch(<ChatParticipantsModal {...overrideProps(propOverrides)} />);
}

function callMapStateToProps(stateOverrides, ownPropsOverrides) {
    const state = Object.assign({}, initialState, stateOverrides || {});
    const ownProps = Object.assign({}, defaultOwnProps, ownPropsOverrides || {});

    return mapStateToProps(state, ownProps);
}

describe('ChatParticipantsModal', () => {
    it('should render correctly', () => {
        assertSnapshotMatch();
    });

    describe('mapStateToProps', () => {
        it('should return expected props', () => {
            const stateOverrides = { roomId: 666 };
            const props = callMapStateToProps(stateOverrides);

            expect(props.chatParticipants).toBe(initialState.chatParticipants);
            expect(props.onCloseRequest).toBe(defaultOwnProps.onCloseRequest);
        });
    });
});
