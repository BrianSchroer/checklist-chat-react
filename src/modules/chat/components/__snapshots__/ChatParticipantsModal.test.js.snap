// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatParticipantsModal should render correctly 1`] = `
<div
  className="checklist-chat-modal-backdrop"
  role="dialog"
  tabIndex="-1"
>
  <div
    className="modal-dialog checklist-chat-modal-dialog"
    role="document"
  >
    <div
      className="modal-content"
    >
      <form
        onSubmit={[Function]}
      >
        <div
          className="modal-header bg-primary text-white"
        >
          <button
            aria-label="Close"
            className="close"
            onClick={[Function]}
            type="button"
          >
            <span
              aria-hidden="true"
            >
              Ã—
            </span>
          </button>
          <h4
            className="modal-title display-linebreaks"
          >
            Who's Here?
          </h4>
        </div>
        <div
          className="modal-body"
        >
          <div
            className="chat-participants"
          >
            <table
              className="table"
            >
              <thead>
                <tr>
                  <th>
                    Name
                  </th>
                  <th>
                    Department
                  </th>
                  <th>
                    Title
                  </th>
                  <th>
                    Connection
                  </th>
                </tr>
              </thead>
              <tbody>
                <tr>
                  <td>
                    name 1
                  </td>
                  <td>
                    department 1
                  </td>
                  <td>
                    title 1
                  </td>
                  <td>
                    connection 1
                  </td>
                </tr>
                <tr>
                  <td>
                    name 2
                  </td>
                  <td>
                    department 2
                  </td>
                  <td>
                    title 2
                  </td>
                  <td>
                    connection 2
                  </td>
                </tr>
                <tr>
                  <td>
                    name 3
                  </td>
                  <td>
                    department 3
                  </td>
                  <td>
                    title 3
                  </td>
                  <td>
                    connection 3
                  </td>
                </tr>
                <tr>
                  <td>
                    name 4
                  </td>
                  <td>
                    department 4
                  </td>
                  <td>
                    title 4
                  </td>
                  <td>
                    connection 4
                  </td>
                </tr>
                <tr>
                  <td>
                    name 5
                  </td>
                  <td>
                    department 5
                  </td>
                  <td>
                    title 5
                  </td>
                  <td>
                    connection 5
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
        <div
          className="modal-footer"
        >
          <input
            className="btn btn-primary"
            onClick={[Function]}
            type="button"
            value="Close"
          />
        </div>
      </form>
    </div>
  </div>
</div>
`;
